### PR TITLE
Add `add --first` option to add the branch as the first in the list

### PR DESCRIPTION
**Pull request by matthalp**
I often find that I care most about recently created branches instead of ones I have hanging around -- both visually and for traversal. It seems like adding new branches first in the list is an easy way to accomplish that.

### CODE REVIEW

**Comment by matthalp** [comment]
@PawelLipski what do you think of this change? If it seems reasonable I will add the tests and address CI issues.

Also, thank you for making this tool. It has been a superpower for me!

**Comment by PawelLipski** [comment]
Nice, makes sense... I'll take a look later in the day or tomorrow 👁️ 

> Also, thank you for making this tool. It has been a superpower for me!

Aww thank you! btw, how did you find out of this tool?

File: completion/git-machete.completion.bash [hunk 1]
```diff
@@ -14,7 +14,7 @@ _git_machete() {
   local shells="bash fish zsh"
 
   local common_opts="--debug -h --help -v --verbose"
-  local add_opts="-o --onto= -R --as-root -y --yes"
+  local add_opts="-o --onto= -R --as-root -y --yes -f --first"
```
Comment (PawelLipski) [inline completion/git-machete.completion.bash]: Pls reorder to keep abc order, here and in other completion files

**Review by PawelLipski — COMMENTED** [review]
Ok! looks good, pls add the tests as we strive to keep >98% coverage (you should a comment from Codecov on this PR) 😅 also, I've changed the base to `develop`, as `master` is for released versions only

**Comment by matthalp** [comment]
> Aww thank you! btw, how did you find out of this tool?

From this [article](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/) that was on HN a long time ago

File: completion/git-machete.completion.bash (hunk 1 above)
Comment (matthalp) [reply completion/git-machete.completion.bash]: Done -- here and the other files!

**Comment by PawelLipski** [comment]
> > Aww thank you! btw, how did you find out of this tool?
> 
> From this [article](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/) that was on HN a long time ago

Heh this article (July 2022) was a seminal moment in [this project's history](https://star-history.com/#VirtusLab/git-machete&Date) :trollface: 

image

I owe this guy a bottle of a very good whisky!

File: completion/git-machete.fish [hunk 2]
```diff
@@ -36,6 +36,7 @@ end
 # git machete add
 complete -c git-machete -n "not __fish_seen_subcommand_from $__machete_commands"                                         -f                 -a add                            -d 'Add a branch to the tree of branch dependencies'
 complete -c git-machete -n "__fish_seen_subcommand_from add"                                                             -f                 -a '(__machete_addable_branches)'
+complete -c git-machete -n "__fish_seen_subcommand_from add; and not __fish_seen_subcommand_from --first -f"             -f -l first   -s f                                   -d 'Add the given branch first in the list of the parent's child branches'
```
Comment (PawelLipski) [inline completion/git-machete.fish]:
```suggestion
complete -c git-machete -n "__fish_seen_subcommand_from add; and not __fish_seen_subcommand_from --as-root -R --first -f"             -f -l first   -s f                                   -d 'Add the given branch first in the list of the parent's child branches'
```

Comment (PawelLipski) [reply completion/git-machete.fish]: As `Option -R/--as-root cannot be specified together with -f/--first`

File: git_machete/cli.py [hunk 3]
```diff
@@ -143,6 +143,7 @@ def create_cli_parser() -> argparse.ArgumentParser:
         add_help=False,
         parents=[common_args_parser])
     add_parser.add_argument('branch', nargs='?')
+    add_parser.add_argument('-f', '--first', action='store_true')
```
Comment (PawelLipski) [inline git_machete/cli.py:146]: Actually... I'd consider renaming the long form to `--as-first-child`:
1. The meaning is clearer
2. The name is more consistent with the existing `--as-root` flag

**Comment by PawelLipski** [comment]
Hey @matthalp... do you have some work done on this one? I can pick it up from the current state and release within v3.24.0

**Comment by matthalp** [comment]
Hi @PawelLipski I was between OOO when I authored this and am out for the next two weeks. Please feel free to finish this if you'd like. I'm sorry I hadn't gotten it done yet.

**Comment by PawelLipski** [comment]
Work to be continued at #1184 (as I can't push to your fork). Your contribution will still count as if this PR has been merged directly.

**Comment by PawelLipski** [comment]
Done, to be released with v3.24.0

**Pull request closed without merging**